django_setup_configuration/documentation/setup_config_usage.py

```diff
@@ -0,0 +1,137 @@
+
+
+def _parse_bool(argument):
+    value = directives.choice(argument, ("true", "false", "yes", "no", "1", "0"))
+    return value in ("true", "yes", "1")
+
+
+@dataclass(frozen=True)
+class StepInfo:
+    title: str
+    description: str
+    anchor_id: str
+    module_path: str
+    step_cls: BaseConfigurationStep
+
+
+class SetupConfigUsageDirective(Directive):
+    has_content = True
+
+    option_spec = {
+        "show_command_usage": _parse_bool,
+        "show_steps": _parse_bool,
+        "show_steps_toc": _parse_bool,
+        "show_steps_autodoc": _parse_bool,
+    }
+
+    def run(self):
+        show_command_usage = self.options.get("show_command_usage", True)
+        show_steps = self.options.get("show_steps", True)
+        show_steps_toc = self.options.get("show_steps_toc", True)
+        show_steps_autodoc = self.options.get("show_steps_autodoc", True)
+
+        if not (settings := self._get_django_settings()):
+            raise ValueError(
+                "Unable to load Django settings. Is DJANGO_SETTINGS_MODULE set?"
+            )
+
+        if not (configured_steps := settings.get("SETUP_CONFIGURATION_STEPS")):
+            raise ValueError(
+                "No steps configured. Set SETUP_CONFIGURATION_STEPS via your "
+                "Django settings."
+            )
+
+        usage_template = self._load_usage_template()
+        steps = self._load_steps(configured_steps)
+
+        rst = ViewList()
+        rst.append("", "<dynamic>")
+        usage_rst = usage_template.render(
+            context=Context(
+                {
+                    "steps": steps,
+                    "show_toc": show_steps_toc,
+                    "show_steps": show_steps,
+                }
+            )
+        )
+        lines = usage_rst.split("\n")
+        for line in lines:
+            rst.append(line, "<dynamic>")
+
+        root_node = nodes.container()
+        if show_command_usage:
+            root_node = nodes.section()
+            root_node["ids"] = ["django-setup-config"]
+            root_node += nodes.title(
+                text="Using the setup_configuration management command"
+            )
+            self.state.nested_parse(rst, 0, root_node)
+
+        if show_steps:
+            for step in steps:
+                rst = ViewList()
+                step_node = nodes.section(ids=[step.anchor_id])
+                step_node += nodes.title(text=step.title)
+
+                if show_steps_autodoc:
+                    rst.append(f".. autoclass:: {step.module_path}", "<dynamic>")
+                    rst.append("    :noindex:", "<dynamic>")
+                else:
+                    # Explicitly display the docstring if there's no autodoc to serve
+                    # as the step description.
+                    for line in step.description.splitlines():
+                        rst.append(line, "<dynamic>")
+
+                rst.append(f".. setup-config-example:: {step.module_path}", "<dynamic>")
```

Comment:
@swrichards something else I ran into just now: it might be useful to add anchors to the steps, that way you can still reference specific steps from other pages

```
.. _setup_configuration_Baz:

.. setup-config-example:: foo.bar.Baz
```

not entirely sure what the name of the reference should be though